File: 2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.cpp
class Solution {
public:
    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        vector<int>ans;
        int n=words.size();
        vector<int>pref(n,0);
        set<char>st={'a','e','i','o','u'};

        int cnt=0;
        for(int i=0;i<n;i++){
           auto it1=st.find(words[i][0]);
           auto it2=st.find(words[i].back());
           if(it1!=st.end() and it2!=st.end()) cnt++;

           pref[i]=cnt;
        }
        
        for(auto it:queries){
         int tmp=pref[it[1]]-pref[it[0]];
         auto it1=st.find(words[it[0]][0]);
           auto it2=st.find(words[it[0]].back());
           if(it1!=st.end() and it2!=st.end()) tmp++;
           ans.push_back(tmp);

        }
        return ans;
    }
};

File: 2691-count-vowel-strings-in-ranges/count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        v={'a','e','i','o','u'}
        n=len(words)
        pref=[0]*n
        cnt=0
        for i in range(len(words)):
            if words[i][0] in v and words[i][-1] in v:
                cnt+=1
            pref[i]=cnt
        ans=[]
        for it in queries:
            lef=it[0]
            rig=it[1]
            if lef==0:
                ans.append(pref[rig])
            else:
                ans.append(pref[rig]-pref[lef-1])
        

        return ans



                
            
      

        